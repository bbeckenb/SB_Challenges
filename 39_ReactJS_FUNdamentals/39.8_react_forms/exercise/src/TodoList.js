import React, { useState } from "react";
import "./TodoList.css"
import { v4 as uuid } from 'uuid';
import Todo from "./Todo";
import NewTodoForm from "./NewTodoForm";

function TodoList() {
    const [listOfTodos, updateTodos] = useState([]) 
    function removeTodo(todoId) {
        updateTodos(todos => todos.filter(t => t.id !== todoId))
    }
    function addTodo(newTodo) {
        updateTodos(todos => [...todos, { ...newTodo, id: uuid() }])
    }


    return (
        <div>
            <h1>Make Todos!</h1>
            <NewTodoForm addTodo={addTodo} />
            <div>
            {listOfTodos.map(({id, color, todo}) => <Todo key={id} id={id} color={color} todo={todo} removeTodo={evt => removeTodo(`${id}`)}/>)}
            </div>
            
        </div>
    )
}

export default TodoList;