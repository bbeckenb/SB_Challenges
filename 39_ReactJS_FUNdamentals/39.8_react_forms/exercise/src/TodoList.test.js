import React from "react";
import {render, fireEvent} from "@testing-library/react";
import TodoList from "./TodoList";

it("renders without crashing", function() {
    render(<TodoList />);
  });
  
  it("matches snapshot", function() {
    const { asFragment } = render(<TodoList />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should add new todo", function () {
    const { queryByText, getByLabelText } = render(<TodoList />);
    const input = getByLabelText("Todo");
    const btn = queryByText("Add Todo!");
    expect(queryByText('make pancakes')).not.toBeInTheDocument();
    fireEvent.change(input, { target: { value: 'make pancakes' } });
    fireEvent.click(btn);
    expect(queryByText('make pancakes')).toBeInTheDocument();
  })

  it("should delete existing todo", function () {
    const { queryByText, getByLabelText } = render(<TodoList />);
    const input = getByLabelText("Todo");
    const btn = queryByText("Add Todo!");
    expect(queryByText('make pancakes')).not.toBeInTheDocument();
    fireEvent.change(input, { target: { value: 'make pancakes' } });
    fireEvent.click(btn);
    expect(queryByText('make pancakes')).toBeInTheDocument();
    const delBtn = queryByText('X');
    fireEvent.click(delBtn);
    expect(queryByText('make pancakes')).not.toBeInTheDocument();
  })