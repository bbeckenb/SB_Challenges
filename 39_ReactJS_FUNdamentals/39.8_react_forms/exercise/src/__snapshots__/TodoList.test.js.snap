// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Make Todos!
    </h1>
    <form>
      <label
        for="todo"
      >
        Todo
      </label>
      <input
        id="todo"
        name="todo"
        type="text"
        value=""
      />
      <label
        for="color"
      >
        Color
      </label>
      <input
        id="color"
        name="color"
        type="color"
        value="#00FF00"
      />
      <button>
        Add Todo!
      </button>
    </form>
    <div />
  </div>
</DocumentFragment>
`;
